packages/datadog-instrumentations/src/graphql.js

```diff
@@ -259,59 +259,26 @@ function pathToArray (path) {
   return flattened.reverse()
 }
 
-function assertField (context, info, args) {
+function assertField (parentCtx, info, args) {
   const pathInfo = info && info.path
 
   const path = pathToArray(pathInfo)
 
   const pathString = path.join('.')
-  const fields = context.fields
+  const fields = parentCtx.fields
 
   let field = fields[pathString]
 
   if (!field) {
-    const parent = getParentField(context, path)
-
-    // we want to spawn the new span off of the parent, not a new async resource
-    parent.asyncResource.runInAsyncScope(() => {
-      /* this child resource will run a branched scope off of the parent resource, which
-      accesses the parent span from the storage unit in its own scope */
-      const childResource = new AsyncResource('bound-anonymous-fn')
-
-      childResource.runInAsyncScope(() => {
-        startResolveCh.publish({
-          info,
-          context,
-          args
-        })
-      })
-
-      field = fields[pathString] = {
-        parent,
-        asyncResource: childResource,
-        error: null
-      }
-    })
-  }
-
-  return field
-}
-
-function getParentField (context, path) {
-  for (let i = path.length - 1; i > 0; i--) {
-    const field = getField(context, path.slice(0, i))
-    if (field) {
-      return field
+    const fieldCtx = { info, ctx: parentCtx, args }
+    startResolveCh.publish(fieldCtx)
+    field = fields[pathString] = {
```

Comment:
you are replacing `context` property by `ctx` here, is the reason why appsec tests are failing.

If you want to maintain that, you have to modify the references of `context` in appsec subscriptions, like here:
https://github.com/DataDog/dd-trace-js/blob/c17708e05dec0fda4992c54065fc267abfbd52e7/packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js#L124

Reply:
Ahh i didn't catch that, thank you for this, just changed it hopefully all the appsec failures dissappear.